libvast/vast/concept/convertible/data.hpp

```diff
@@ -70,7 +70,7 @@ caf::error insert_to_map(To& dst, typename To::key_type&& key,
     else
       // TODO: Consider continuing if the old and new values are the same.
       return caf::make_error(ec::convert_error,
-                             fmt::format("{}: redefinition detected: \"{}\" "
+                             fmt::format(": redefinition of {} detected: \"{}\" "
```

Comment:
```suggestion
                             fmt::format("redefinition of {} detected:  {} "
```

Why not this?

Reply:
I am not a fan of beginning the error context with a colon either. If you want to make this a thing @tobim then please make it consistently throughout the code base by adding the colon in the error rendering function.

Reply:
I think this is not even an option, because it destroys composition.

Reply:
The data conversion overloads are a special case because the message will almost always get prepended with the path that leads to the location of the error. The desired output should be
```
convert_error: foo.bar[2].baz: redefinition of key detected: "A" vs "B"
```
I'm open to ideas for how to implement that without the need to add the colon to the inner message.

Reply:
You can add the colon when rendering nested errors, just like you special-case std::string in error contexts now.